File: Foundation/Arrays/lastindex.java
 import java.util.Scanner;
public class lastindex
{
    public static Scanner scn = new Scanner(System.in);
    public static int lastIndex(int[] arr)
    {
        int num = scn.nextInt();
        int index = -1;
        for(int i = arr.length-1; i >=0; i--)
        {
            if(arr[i] == num)
            {
                index = i;
                break;
            }
        }
        return index;
    }
    public static void main(String[] args)
    {
        int n = scn.nextInt();
        int[] arr = new int[n];
        for(int i = 0; i < arr.length; i++)
        {
            arr[i] = scn.nextInt();
        }
        System.out.println(lastIndex(arr));
    }
}

File: Foundation/Arrays/lastindex.cpp
#include<iostream>
#include<vector>
using namespace std;

int lastIndex(vector<int>& arr)
{
    int num;
    cin>>num;
    int index = -1;
    for(int i = arr.size(); i > 0; i--)
    {
        if(arr[i] == num)
        {
            index = i;
            break;
        }
    }
    return index;
}

int main()
{
    int n;
    cin>>n;
    vector<int> arr(n);
    for(int i = 0; i< arr.size(); i++)
    {
        cin>>arr[i];
    }
    cout<<lastIndex(arr);
    return 0;
}